### DISCUSSION TITLE
Call SolidJS onMount and onCleanup each time Modal is .show or .hide

### DISCUSSION THREAD

**Discussion by robodna** [mdbootstrap/TW-Elements#2144] [Chat about dev stuff]
I am using SolidJS with the Modal component.

The onMount and onCleanup methods are only called once, and I would like them to be called each time the Modal is shown or hidden.  Is this possible?

Thank you

**Comment by juujisai** [comment]
The modal is emitting few events that you can listen to and call the methods used inside the `onMount` and `onCleanup`. Try listening for `shown.te.modal` and `hidden.te.modal` events. 

Checkout our Events section for Modals here:
https://tw-elements.com/docs/standard/components/modal/#api-section-events

Hope it helps!
